xblockutils/templatetags/i18n.py

```diff
@@ -0,0 +1,49 @@
+"""
+Template tags for handling i18n translations for xblocks
+
+Based on: https://github.com/eduNEXT/django-xblock-i18n
+"""
+from django.template import Library, Node
+from django.templatetags.i18n import do_translate, do_block_translate
+
+
+register = Library()
+
+
+class ProxyTransNode(Node):
+    """
+    This node is a proxy of a django TranslateNode.
+    """
+    def __init__(self, do_translate_node):
+        """
+        Stores the provided "do_translate" node
+        """
+        self.do_translate = do_translate_node
+
+    def render(self, context):
+        """
+        Renders the translated text using the XBlock i18n service, if available.
+        """
+        django_translated = self.do_translate.render(context)
+
+        i18n_service = context.get('_i18n_service', None)
+        if i18n_service:
+            return i18n_service.gettext(django_translated)
```

Comment:
I guess this does not support things such as `{% trans "my string" as variable %}` or [`blocktrans` features](https://docs.djangoproject.com/en/2.0/topics/i18n/translation/#blocktrans-template-tag) such as variable substitution or support for plural forms. I'm not sure how easy it would be to support these, though. If it's hard, I think a proxy like this is fine for the initial implementation.

Reply:
Oo good point -- I dug a bit deeper and found a way to do that which should continue to work through Django 2.0. See b9c2ae2.